### ISSUE TITLE
Docker-built binary fails with "error decoding response body"

### ISSUE THREAD

**Issue by LRNZ09** [Caleb9/syno-photo-frame#18]
## Environment
- **Device**: Raspberry Pi 4
- **Build Command**: `docker build . --target=bin --output type=local,dest=.`

## Issue Description
The binary built from the source code using Docker is failing immediately after startup with a login error, despite using publicly shared album links.

## Error Message
```bash
2024-09-11T20:27:23.197Z ERROR [syno_photo_frame] error decoding response body
Error: "Login to Synology Photos failed. Make sure the share link is pointing to a *publicly shared album*. If the album's password link protection is enabled, use the --password option with a valid password."
```

## Steps to Reproduce
1. Build the binary using Docker as per instructions: `docker build . --target=bin --output type=local,dest=.`
2. Run the resulting `syno_photo_frame` binary
3. Observe immediate crash with the above error message

## Expected Behavior
The program should successfully log in and access the publicly shared Synology Photos albums.

## Actual Behavior
The program crashes immediately with a login error, suggesting it cannot access the shared albums.

## Additional Information
- I tried with two different types of publicly shared album links:
  1. A standard album with just one photo
  2. A conditional album using a certain tag
- Both types of links result in the same error
- They are freely accessible without password

## Questions
1. Are there any other specific requirements or configurations needed for the shared links that might not be obvious?
2. Could there be an issue with how the binary is interacting with the Synology Photos API on this particular setup?

Any assistance or guidance on resolving this issue would be greatly appreciated. Thank you!

**Comment by Caleb9** [comment]
Hi @LRNZ09, are you able to test if the same occurs if binary is built without Docker, so we could eliminate that as the possible reason? Following [this](https://github.com/Caleb9/syno-photo-frame?tab=readme-ov-file#option-2-build-from-source) it's a 3 commands operation. Or try the [pre-built .deb package](https://github.com/Caleb9/syno-photo-frame/releases/download/0.13.0/syno-photo-frame_0.13.0_arm64.deb).

Aside from that which DSM, Synology Photos and RPi OS versions, and RPi model are you using?

**Comment by LRNZ09** [comment]
Hi @Caleb9, thanks for your quick feedback!

I've conducted further tests as you suggested, and here are the results:

1. I tried both building from source without Docker and installing the pre-built .deb package. Unfortunately, the same error occurred in both cases.

2. Regarding the Synology Photos link, I'm using the link provided by the Synology Photos sharing modal, which is in the format `http://***.quickconnect.to/mo/sharing/***`. 
<img src="https://github.com/user-attachments/assets/3a5493cd-4a05-41ed-96ee-bb9cf7915151" height=400 />


3. Interestingly, when I try using the local hostname with HTTPS, the error changes to: `error sending request for url (https://synologynas:5001/mo/sharing/webapi/entry.cgi)`, but when using plain HTTP (`http://synologynas:5000/mo/sharing/***`), no error is shown and it simply hangs forever.

Here are the version details you requested:
   - DSM version: DSM 7.2.1-69057 Update 5
   - Synology Photos version: 1.6.2-0710

I'm running this on a Raspberry Pi 4 Model B Rev 1.2. Here's the detailed system information:

   ```
          _,met$$$$$gg.          pi@pi
       ,g$$$$$$$$$$$$$$$P.       -----
     ,g$$P"     """Y$$.".        OS: Debian GNU/Linux 12 (bookworm) aarch64
    ,$$P'              `$$$.     Host: Raspberry Pi 4 Model B Rev 1.2
   ',$$P       ,ggs.     `$$b:   Kernel: 6.6.31+rpt-rpi-v8
   `d$$'     ,$P"'   .    $$$    Uptime: 48 mins
    $$P      d$'     ,    $$P    Packages: 1638 (dpkg)
    $$:      $$.   -    ,d$$'    Shell: bash 5.2.15
    $$;      Y$b._   _,d$P'      Resolution: 2560x1440
    Y$$.    `.`"Y$$$$P"'         Terminal: /dev/pts/0
    `$$b      "-.__              CPU: (4) @ 1.500GHz
     `Y$$                        Memory: 738MiB / 3792MiB
      `Y$$.
        `$$b.
          `Y$$b.
             `"Y$b._
                 `"""
   ```

So it seems the issue persists regardless of the build method. The difference in behavior between HTTPS and HTTP connections might be a clue. Could there be an issue with SSL/TLS handling or certificate validation?

**Comment by Caleb9** [comment]
@LRNZ09 thanks for the details. So QuickConnect is currently not supported, but I think you're close to the solution. It seems your HTTP(S) links are missing the path part for Photos app. The `https://synologynas:5001` URL points to DSM "desktop". If you're using an alias (Control Panel -> Login Portal -> Applications -> Synology Photos -> Alias) e.g. "photo", that link would be https://synologynas:5001/photo/mo/sharing/ABC1234 (note the "photo" part in the path). Maybe give it a try and let me know if it helped, and if you don't want to use an alias we'll dig deeper ;).

**Comment by LRNZ09** [comment]
Ok, I've tested the setup with the alias `photo` as you suggested. I then tried using a link in the format: `http://synologynas.local/photo/mo/sharing/ABC1234`, which works as expected when visited in a browser, with no redirects involved.

Unfortunately, the issue persists, and the command continues to hang indefinitely. There's no verbose mode available in the binary, so I couldn't investigate further or understand what's happening under the hood.

**Comment by Caleb9** [comment]
You can add `RUST_LOG=debug` or `RUST_LOG=trace` before the command, so e.g. `RUST_LOG=debug syno-photo-frame https://...` to enable debugging. Let me know what you find out, thanks.

**Comment by LRNZ09** [comment]
I spent a few hours analyzing the code, even though I'm not very experienced with Rust. I compared it to how my version of the Synology Photos app calls the APIs, and it looks like they changed it to use a different endpoint, `http://synologynas.local/photo/synofoto/api/v2/p/METHOD_NAME/get`. Besides that, I tried with the `RUST_LOG=trace` and it seemed like a problem with SDL getting stuck in a thread. Luckily a reboot solved the issue, and now it works as expected 😅
